website/pages/[handle]/_/access-tokens.tsx

```diff
@@ -0,0 +1,247 @@
+import H3 from '~/components/ui/typography/H3'
+import CopyButton from '~/components/ui/CopyButton'
+import { useDeleteAccessTokenMutation } from '~/data/access-tokens/delete-access-token'
+
+const ApiTokensPage: NextPageWithLayout = () => {
+  const router = useRouter()
+  const { handle } = useParams()
+  const { refreshSession } = useAuth()
+
+  const [showNewTokenForm, setShowNewTokenForm] = useState(false)
+  const [newToken, setNewToken] = useState('')
+
+  const { mutateAsync: createNewAccessToken, isLoading: creatingNewAccessToken } = useNewAccessTokenMutation({
+    onSuccess() {
+      toast.success('Successfully created token!')
+    },
+  })
+
+  const {
+    data: accessTokens,
+    isLoading: accessTokensLoading,
+    isSuccess: isAccessTokensSuccess,
+  } = useAccessTokensQuery()
+
+  const createNewToken = async ({ tokenName }: { tokenName: string }) => {
+    try {
+      await refreshSession()
+      const newToken = await createNewAccessToken({ tokenName })
+      setNewToken(newToken)
+    } catch (error: any) {
+      const message =
+        error.message ??
+        `Sorry, we had an unexpected error. Please try again. - ${error.toString()}`
+      return {
+        [FORM_ERROR]: message,
+      }
+    }
+  }
+
+  const { mutateAsync: deleteAccessToken } = useDeleteAccessTokenMutation({
+    onSuccess() {
+      toast.success('Successfully revoked token!')
+      router.replace(`/${handle}/_/access-tokens`)
+    },
+  })
+
+  const revokeToken = async (tokenId: string) => {
+    try {
+      deleteAccessToken({ tokenId })
+    } catch (error: any) {
+      return {
+        [FORM_ERROR]:
+          'Sorry, we had an unexpected error. Please try again. - ' +
+          error.toString(),
+      }
+    }
+  }
+
+  type AccessTokensPageProps = {
+    onClickNewToken: MouseEventHandler<HTMLButtonElement>
+    isSuccess: boolean
+    isLoading: boolean
+    accessTokens: AccessTokensResponse | undefined
+  }
+
+  const AccessTokensPage = ({
+    onClickNewToken,
+    isSuccess,
+    isLoading,
+    accessTokens,
+  }: AccessTokensPageProps) => (
+    <>
+      <H1 className="!text-3xl">Access Tokens</H1>
+      <div className="flex flex-row my-6">
+        <Button
+          className="bg-slate-700 text-white hover:bg-slate-500 dark:bg-slate-50 dark:text-slate-900 dark:hover:bg-slate-500"
+          onClick={onClickNewToken}
+        >
+          New Token
+        </Button>
+      </div>
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : isSuccess ? (
+        accessTokens &&
+        accessTokens.map((accessToken) => (
+          <AccessTokenCard
+            key={accessToken.id}
+            tokenId={accessToken.id}
+            tokenName={accessToken.token_name}
+            createdAt={accessToken.created_at}
+            onRevokeButtonClick={revokeToken}
+          ></AccessTokenCard>
+        ))
+      ) : (
+        <p className="text-red-500">Error loading access tokens</p>
+      )}
```

Comment:
react-query is smart enough to make all of these options mutually exclusive so we can avoid the ternaries here :)
For example:
```tsx
<>
  {isLoading && <p>Loading...</p>}
  {isSuccess && ...}
  {isError && <p className="text-red-500">Error loading access tokens</p>}
</>
```

Reply:
Didn't know that, thank you. Updated.